crates/consensus/common/src/validation.rs

```diff
@@ -848,4 +848,78 @@ mod tests {
             }))
         );
     }
+
+    #[test]
+    fn test_valid_gas_limit_increase() {
+        let parent = SealedHeader {
+            header: Header { gas_limit: 1024 * 10, ..Default::default() },
+            ..Default::default()
+        };
+        let child = SealedHeader {
+            header: Header { gas_limit: parent.header.gas_limit + 5, ..Default::default() },
+            ..Default::default()
+        };
+        let chain_spec = ChainSpec::default();
+
+        assert_eq!(check_gas_limit(&parent, &child, &chain_spec), Ok(()));
+    }
+
+    #[test]
+    fn test_london_hardfork_adjustment() {
+        let parent = SealedHeader {
+            header: Header { gas_limit: 1024 * 10, ..Default::default() },
+            ..Default::default()
+        };
+        let child = SealedHeader {
+            header: Header { gas_limit: parent.header.gas_limit, ..Default::default() },
+            ..Default::default()
+        };
+        let mut chain_spec = ChainSpec::default();
+        chain_spec.hardforks.insert(Hardfork::London, ForkCondition::Block(1));
+
+        assert_eq!(check_gas_limit(&parent, &child, &chain_spec), Ok(()));
```

Comment:
this probably won't test anything related to london since parent and child both have block number zero in this case, I would set the block number of the parent, and have London fork at block zero when testing block number edge cases